### ISSUE TITLE
downloaded pretrained model and number of training images and converged loss value

### ISSUE THREAD

**Issue by JohnTargaryen** [shepnerd/inpainting_gmcnn#55]
Greetings! Thank you for your excellent work.
I have come across three problems and hope that you or the github community can provide me some insights.

1.  I downloaded the pretrained model paris_streetview provided in the main page, which I unzipped and get the folder below
![image](https://user-images.githubusercontent.com/17902928/75419149-2d555e00-5970-11ea-9eec-4b04f930d2a6.png)

However , when i run the command to test this pretrained model as instructed
```py
python test.py --dataset paris_streetview --data_file ./test.txt --load_model_dir ./checkpoints/paris-streetview_256x256_rect --random_mask 0
```
An error would occur:
![image](https://user-images.githubusercontent.com/17902928/75419265-6c83af00-5970-11ea-9de9-b9b64fec305f.png)

I wonder if the model i downloaded is somehow damaged?(If so, could you kindly send a pretrained model to my e-mail 294484601@qq.com for that it's hard for me to access google drive) Or that I've done something wrong elsewhere?

2. How many images are there in your training datasets? What's the  optimal number of training image? I tried to train the network on Adobe FiveK dataset that contains 5000 images of various scenes and portraits. Is that too many or too few training images?

3. To what value should the loss converge on the pretrain phase and fine-tune phase respectively?

Your prompt response is highly appreciated. Thank you for your time, and wish you all the best!

**Comment by shepnerd** [comment]
You mentioned pretrained models are for tensorflow instead of pytorch. You can try it with tensorflow testing code.

**Comment by JohnTargaryen** [comment]
> You mentioned pretrained models are for tensorflow instead of pytorch. You can try it with tensorflow testing code.

Thank you for your quick reply. I will download the pytorch pretrained model instead.
Also, could you please share some thoughts on the other two questions？ Thanks!

**Comment by shepnerd** [comment]
For your other two questions,

2) It is hard to determine the optimal training images. The basic rule is, the required training number is relatively small when the data are well-aligned (e.g., aligned human faces in celebA), while that number should be numerous for a data distribution containing various categories (e.g., Places2). The training numbers for each used dataset is listed as:
![image](https://user-images.githubusercontent.com/5066504/75513753-b9c75580-5a30-11ea-9575-ef73a163392c.png)
As far as I know, image contents in Adobe5K are quite diverse and 5k seems fewer than what we expect. But, increasing training iterations may help considering image size in Adobe5k is huge and randomly patch sampling may cover the real data distribution when having enough samples.

3) I may check the number later. For a relatively small dataset (e.g., celebA-HQ or Paris streetview), 20-40 training epochs are enough for the pre-training stage. Such figures also work for the fine-tune stage.

**Comment by JohnTargaryen** [comment]
> For your other two questions,
> 
> 1. It is hard to determine the optimal training images. The basic rule is, the required training number is relatively small when the data are well-aligned (e.g., aligned human faces in celebA), while that number should be numerous for a data distribution containing various categories (e.g., Places2). The training numbers for each used dataset is listed as:
>    ![image](https://user-images.githubusercontent.com/5066504/75513753-b9c75580-5a30-11ea-9575-ef73a163392c.png)
>    As far as I know, image contents in Adobe5K are quite diverse and 5k seems fewer than what we expect. But, increasing training iterations may help considering image size in Adobe5k is huge and randomly patch sampling may cover the real data distribution when having enough samples.
> 2. I may check the number later. For a relatively small dataset (e.g., celebA-HQ or Paris streetview), 20-40 training epochs are enough for the pre-training stage. Such figures also work for the fine-tune stage.

Thank you! It helps a lot!
